doc/operations/certificates.rst

```diff
@@ -78,3 +78,48 @@ The procedure that operators and members should follow is summarised in the foll
     section Service <br> Certificate
     Initial Validity Period (24h default): done, 01-01/00:00, 1d
     Post Service Open Validity Period    : 01-01/15:00, 5d
+
+
+ACME-endorsed TLS certificates
+==================================
+
+Unendorsed, self-signed (CA) service certificates are a complication for clients as they need to be given a copy of the certificate before they can establish TLS connections to the service, or the service certificate is permanently installed in their trust store. To alleviate this, CCF provides an `ACME <https://en.wikipedia.org/wiki/Automatic_Certificate_Management_Environment>`_ client, which is used to obtain TLS certificates that are endorsed by external certificate authorities. For instance, the `Let's Encrypt <https://letsencrypt.org/>`_ CA is endorsed by a root certificate that is pre-installed on most current operating systems, which means that clients usually have all required certificates to establish TLS connections without further configuration, if the service certificate is endorsed by Let's Encrypt. CCF handles the creation and renewal of ACME certificates, but it requires some configuration:
+
+1. Get a globally reachable DNS name for your CCF network, e.g. ``my-ccf.example.com``, which resolves to the address of at least one node in the network. Multiple nodes or a load balancer address are fine too.
+
+2. ACME `http-01 <https://letsencrypt.org/docs/challenge-types/>`_ challenges require a challenge server to run on port 80 (non-negotiable). To be able to bind to that port, the ``cchost`` binary may need to be given special permission, e.g. by running ``sudo setcap CAP_NET_BIND_SERVICE=+eip cchost``.
+
+3. Each interface defined in the ``chost`` configuration file can be given the name of an ACME configuration to use. The settings of each ACME configuration are defined in ``network.acme``. Note that this information is required by *all* nodes as they might have to renew the certificate(s) later.
+
+    The various options are as follows:
+
+    .. code-block:: json
+
+        "network": {
+            "rpc_interfaces": { 
+                ... ,
+                "acme_endorsed_interface": { ... , "acme_configuration": "my-acme-cfg" }
+            },
+            "acme": {
```

Comment:
I feel like this could be under `rpc_interfaces` as we have no plan to do this for node-to-node interfaces.

Reply:
The C++ class for node-to-node and RPC interfaces is the same, we would have to split that as well then.

Reply:
You can decide which fields are allowed in the configuration schema, but still have the same structure for both (see https://microsoft.github.io/CCF/main/operations/configuration.html#network).

Reply:
The `acme_configuration` has since moved into the `endorsement`. Is there a way to express that some field of a member must be absent?

Reply:
Just saw that those are copies in the schema instead of a common base definitions, so yes, `endorsement` is already guaranteed to be absent for node-to-node interfaces.